lib/hsa/mcwamp_hsa.cpp

```diff
@@ -1494,22 +1486,29 @@ class HSAQueue final : public KalmarQueue
     bool isEmpty() override {
         // Have to walk asyncOps since it can contain null pointers (if event is waited on and removed)
         // Also not all commands contain signals.
-        for (int i = 0; i < asyncOps.size(); ++i) {
-            if (asyncOps[i] != nullptr) {
-                auto &asyncOp = asyncOps[i];
-                hsa_signal_t signal = *(static_cast <hsa_signal_t*> (asyncOp->getNativeHandle()));
-                if (signal.handle) {
-                    hsa_signal_value_t v = hsa_signal_load_relaxed(signal);
-                    if (v != 0) {
-                        return false;
-                    }
-                } else {
-                    // no signal, have to assume the command is still running
-                    return false;
+        
+        bool isEmpty = true;
+
+        const auto& oldest = find_if(
+                    asyncOps.crbegin(), asyncOps.crend(), [](const std::shared_ptr<HSAOp> &asyncOp) { return asyncOp != nullptr; });
+
+
+        if (oldest != asyncOps.crend()) {
+            hsa_signal_t signal = *(static_cast <hsa_signal_t*> ((*oldest)->getNativeHandle()));
+            if (signal.handle) {
+                hsa_signal_value_t v = hsa_signal_load_acquire(signal);
```

Comment:
Unless I am mistaken, this is deprecated, with `hsa_signal_load_scacquire` being the current / preferred form.